docs/diagnostics/overview.md

```diff
@@ -2,97 +2,232 @@
+
+#### The project file
+
+Using the `NoWarn` property in the project file is great when you need to suppress a diagnostic across the entire project. This is helpful in cases where a diagnostic is irrelevant to your scenario or when dealing with known false positives. It’s a simple, central way to silence a diagnostic without touching your code. However, it lacks visibility—developers won’t see that a diagnostic is suppressed unless they inspect the project file—so it can make it harder to track intentional exceptions or identify the reason behind suppressions.
+
+#### Preprocessor directives
+
+Preprocessor directives like `#pragma warning disable` offer fine-grained suppression within a specific scope, such as a method body or block of code. They're especially handy when you need a temporary suppression during refactoring, or when a rule incorrectly flags a particular line that you can't or don't want to change. The ability to tightly wrap just the affected code makes this approach powerful, but it can also make the code harder to read and maintain, especially if the directives are scattered or forgotten over time.
+
+#### The editor configuration
+
+Suppressing diagnostics using an _.editorconfig_ file is ideal for enforcing or adjusting analyzer behavior at scale. It allows teams to standardize severity levels (or disable rules) across an entire solution or per directory/file pattern. This method keeps suppression cleanly out of the codebase, and it works well for communicating team-wide conventions. However, it can be a bit opaque—developers need to know to look in the config file—and it doesn't offer the precision of an attribute or pragma when dealing with one-off cases.
+
+### Summary of suppression methods
+
+When deciding how to suppress a diagnostic, consider the scope of the suppression, the visibility you want for the suppression, and the best use case for each method. Here's a summary of the methods available:
+
+| Method                    | Scope                 | Best for                        | Visibility           |
+|---------------------------|-----------------------|---------------------------------|----------------------|
+| `[SuppressMessage]`       | Localized             | Documented exceptions           | High (inline)        |
+| `NoWarn` in _.csproj_     | Project               | Global or known false positives | Low                  |
+| `#pragma warning disable` | File/block/line       | Temporary or block-based issues | Medium (inline)      |
+| _.editorconfig_           | Directory             | Consistent team-wide settings   | Medium (centralized) |
+
+### Suppress with attribute
```

Comment:
It feels like these sections and the above sections should be combined. We list them with commentary above, and give examples here.